aten/src/ATen/native/cuda/vol2col.cuh

```diff
@@ -104,7 +104,7 @@ void vol2col(
     T* data_col) {
   // We are going to launch channels * depth_col * height_col * width_col
   // kernels, each kernel responsible for copying a single-channel grid.
-  int num_kernels = channels * depth_col * height_col * width_col;
+  int64_t num_kernels = channels * depth_col * height_col * width_col;
```

Comment:
I believe you need to cast one of the operands into int64_t first otherwise the overflowed int32 will be assigned to the `num_kernels` value.

also leaving variable type as `auto` is sufficient if you did the operand cast

Reply:
Ah, good catch. I've added the cast.